### PR TITLE
check domain permissions before sending SMS

### PR DESCRIPTION
**Pull request by nickpell**
fixes (2) in http://manage.dimagi.com/default.asp?154961#861270

@gcapalbo @biyeun

### CODE REVIEW

**Comment by dimagimon** [comment]
:+1: No lint errors found.

File: corehq/apps/sms/api.py [hunk 1]
```diff
@@ -182,6 +182,12 @@ def send_message_via_backend(msg, backend=None, orig_phone_number=None):
     except Exception:
         logging.exception("Could not clean text for sms dated '%s' in domain '%s'" % (msg.date, msg.domain))
     try:
+        if not domain_has_privilege(msg.domain, privileges.OUTBOUND_SMS):
+            raise Exception(
```
Comment (gcapalbo) [inline corehq/apps/sms/api.py]: Would be nice if we used a separate Exception subclass (something like MissingPrivilegeException), but I don't feel too strong about that.

**Comment by gcapalbo** [comment]
Looks good to me, we just need to notify some people before merging this. I'll merge this once we're done with that.

**Comment by nickpell** [comment]
re-opening to run travis tests

**Comment by nickpell** [comment]
re-closing

**Comment by nickpell** [comment]
@biyeun Can you explain why the tests are failing?  I'm inheriting from ```BaseAccountingTest```

**Comment by nickpell** [comment]
re-opening to run travis tests again (thanks @biyeun for the fix!)

**Comment by dimagimon** [comment]
:+1: No lint errors found.

**Comment by dimagimon** [comment]
:+1: No lint errors found.

**Comment by nickpell** [comment]
re-closing

**Comment by nickpell** [comment]
re-opening to run travis tests

**Comment by dimagimon** [comment]
There are 11 errors:

* corehq/apps/sms/tests/__init__.py, line 4 - F401 'BackendAuthorizationException' imported but unused
* corehq/apps/sms/tests/__init__.py, line 4 - E501 line too long (151 > 115 characters)
* corehq/apps/sms/tests/__init__.py, line 58 - E261 at least two spaces before inline comment
* corehq/apps/sms/tests/__init__.py, line 60 - E231 missing whitespace after ','
* corehq/apps/sms/tests/__init__.py, line 69 - E231 missing whitespace after ','
* corehq/apps/sms/tests/__init__.py, line 72 - E231 missing whitespace after ','
* corehq/apps/sms/tests/__init__.py, line 75 - E231 missing whitespace after ','
* corehq/apps/sms/tests/__init__.py, line 75 - E501 line too long (125 > 115 characters)
* corehq/apps/sms/tests/__init__.py, line 78 - E231 missing whitespace after ','
* corehq/apps/sms/tests/__init__.py, line 80 - W293 blank line contains whitespace
* corehq/apps/sms/tests/__init__.py, line 81 - E231 missing whitespace after ','

File: corehq/apps/reminders/tests/__init__.py [hunk 2]
```diff
@@ -2,6 +2,15 @@
...
 from casexml.apps.case.models import CommCareCase
 from casexml.apps.case.sharedmodels import CommCareCaseIndex
+from corehq.apps.accounting.models import (
+    BillingAccount,
+    DefaultProductPlan,
+    SoftwarePlanEdition,
+    Subscription,
+    SubscriptionAdjustment,
+)
+from corehq.apps.accounting.tasks import activate_subscriptions
```
Comment (dimagimon) [inline corehq/apps/reminders/tests/__init__.py]: F401 'activate_subscriptions' imported but unused

File: corehq/apps/sms/tests/__init__.py [hunk 3]
```diff
@@ -48,8 +55,10 @@ def invoke_doc_exists(self):
         except ResourceNotFound:
             return False
 
-class BackendTestCase(TestCase):
+class BackendTestCase(BaseAccountingTest):
```
Comment (dimagimon) [inline corehq/apps/sms/tests/__init__.py]: E302 expected 2 blank lines, found 1

File: corehq/apps/sms/tests/__init__.py [hunk 4]
```diff
@@ -202,6 +229,7 @@ def test_backend(self):
 
         # Test overriding with a domain-level backend
 
+        self.domain_obj = Domain.get(self.domain_obj._id) # Prevent resource conflict
```
Comment (dimagimon) [inline corehq/apps/sms/tests/__init__.py]: E261 at least two spaces before inline comment

**Comment by nickpell** [comment]
ignoring pre-existing formatting issues raised by dimagimon

**Comment by dimagimon** [comment]
:+1: No lint errors found.

**Comment by nickpell** [comment]
re-closing

**Comment by nickpell** [comment]
@gcapalbo Wondering if this can be merged yet (trying to wind down my backlog today)

**Comment by gcapalbo** [comment]
@nickpell Unfortunately, not yet, still waiting to hear back from one person. Will keep you posted

**Comment by nickpell** [comment]
ok thanks!

**Comment by nickpell** [comment]
@gcapalbo Any closer on this one?

**Comment by gcapalbo** [comment]
@nickpell still working on sorting it out - there ended up being a lot more than i thought

**Comment by nickpell** [comment]
ok, just checking :)

**Comment by gcapalbo** [comment]
@nickpell good news, we can finally merge this - but looks like there's some conflicts with master

**Comment by nickpell** [comment]
:white_check_mark:

**Pull request merged**